PyStemmusScope/soil_io.py

```diff
@@ -0,0 +1,229 @@
+    """Internal function that reads the fmax variable from the surface dataset and
+    returns it in a dictionary.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the surface data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the `fmax` value (maximum fractional saturated area)
+    """
+    ds = xr.open_dataset(soil_data_path / 'surfdata.nc')
+    ds = ds.sel(
+        lsmlat=vc.lat_to_lsmlat(lat), lsmlon=vc.lon_to_lsmlon(lon))
+
+    fmax = ds['FMAX'].values
+
+    surface_matfiledata = {'fmax': fmax}
+
+    return surface_matfiledata
+
+
+def _collect_soil_data(soil_data_path, lat, lon):
+    """Internal function that calls the individual data collectors and merges them into
+    a single dictionary ready to be written.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the soil data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing all the processed soil property data.
+    """
+
+    matfiledata = {}
+
+    lambda_directory = soil_data_path / 'lambda'
+
+    matfiledata.update(_read_lambda_coef(lambda_directory, lat, lon))
+    matfiledata.update(_read_hydraulic_parameters(soil_data_path, lat, lon))
+    matfiledata.update(_read_soil_composition(soil_data_path, lat, lon))
+    matfiledata.update(_read_surface_data(soil_data_path, lat, lon))
+
+    return matfiledata
+
+
+def _retrieve_latlon(plumber2_file):
+    """Retrieves the latitude and longitude coordinates from the PLUMBER2 dataset file,
+    to allow for the
+
+    Args:
+        plumber2_file (Path): Full path to the netCDF file containing the site latitude
+            and longitude
+
+    Returns:
+        tuple(float, float): Tuple containing the latitude and longitude values.
+            Latitude in degrees N, longitude in degrees E.
+    """
+    ds = xr.open_dataset(plumber2_file)
+    ds = ds.squeeze()
+    lat = float(ds.latitude.values)
+    lon = float(ds.longitude.values)
+    return lat, lon
+
+
+def prepare_soil_data(soil_data_dir, matfile_path, run_config):
+    """Function that prepares the soil input data for the STEMMUS_SCOPE model. It parses
+    the data for the input location, and writes a file that can be easily read in by
+    Matlab.
+
+    Args:
+        soil_data_dir (Path): Path to the directory which contains the soil data.
+        mathfile_path (Path): Path to the directory where soil parameter file
+            should be written to.
+        run_config (dict): Dictionary containing the configuration for the current
+            STEMMUS_SCOPE run.
+    """
+    forcing_file = Path(run_config['ForcingPath']) / run_config['ForcingFileName']
+    lat, lon = _retrieve_latlon(forcing_file)
```

Comment:
We should add the fix for missing values at ID-Pag site as implemented in [soilpropertyread.m](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/fbec2c67397952b219c5e430cf76cb8f9df53d55/src/soilpropertyread.m#L4) to here together with #ToDo fix issue https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/77.